test/e2e/config_test.go

```diff
@@ -689,6 +690,121 @@ rules:
 	})
 }
 
+func TestConfigHavingRegex(t *testing.T) {
+	yaml1 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+  annotations:
+    foo1: bar1
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+  annotations:
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+`
+
+	yaml2 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+data:
+  player_initial_lives: "3"
+`
+
+	configYaml := `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+  - path: [metadata, annotations, {regex: "^foo"}]
+    type: %s
+    sources: [new, existing]
+    resourceMatchers:
+      - apiVersionKindMatcher: {apiVersion: v1, kind: ConfigMap}
+`
+
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	fieldsExcludedInMatch := []string{"kapp.k14s.io/app", "creationTimestamp:", "resourceVersion:", "uid:", "selfLink:", "kapp.k14s.io/association"}
+	name := "test-config-path-regex"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy configmaps with annotations", func() {
+		_, _ = kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--diff-changes-yaml"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+	})
+
+	logger.Section("deploy configmaps without annotations", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--diff-changes-yaml"}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml2 + fmt.Sprintf(configYaml, "copy"))})
+		expectedOutput := ``
+
+		out = strings.TrimSpace(replaceTarget(replaceSpaces(replaceTs(out))))
+		out = clearKeys(fieldsExcludedInMatch, out)
+
+		expectedOutput = strings.TrimSpace(replaceSpaces(expectedOutput))
+		// rebase rule should copy the annotations matched with regex
+		require.Contains(t, out, expectedOutput, "output does not match")
```

Comment:
Should I print the ConfigMaps with kubectl?

Reply:
Or should I use some tool to just print the annotations of both the configmaps?

Reply:
No new tool is needed. you can use kubectl only. It has provision to print only things you are interested in. e.g.
```kubectl get cm/kapp-controller-config -n kapp-controller -o jsonpath='{.metadata.annotations}'```

Reply:
I think the fact that there is no diff should be fine to show that foo* got copied. Although, let's use `--diff-run` and `--diff-exit-status` like [this](https://github.com/vmware-tanzu/carvel-kapp/blob/1ecc9149f578d6e700b27cea4b28cac70ca32a16/test/e2e/config_test.go#L616) instead of `--diff-changes-yaml` to make it more clean.

Reply:
Done